src/docfx/build/toc/parser/TableOfContentsParser.cs

```diff
@@ -21,10 +21,19 @@ internal static class TableOfContentsParser
 
         public delegate (string content, Document file) ResolveContent(Document relativeTo, SourceInfo<string> href, bool isInclusion);
 
+        public delegate List<string> ResolveMoniker(Document doc);
+
         public static (List<Error> errors, TableOfContentsModel model)
-            Load(Context context, Document file, ResolveContent resolveContent, ResolveHref resolveHref, ResolveXref resolveXref)
+            Load(Context context, Document file, ResolveContent resolveContent, ResolveHref resolveHref, ResolveXref resolveXref, ResolveMoniker resolveMoniker)
         {
-            return LoadInternal(context, file, file, resolveContent, resolveHref, resolveXref, new List<Document>());
+            var errors = new List<Error>();
+            var (monikerError, fileMonikers) = context.MonikerProvider.GetFileLevelMonikers(file);
```

Comment:
will directly use `resolveMoniker` here?

Reply:
sure

---
In reply to: [293179060](https://github.com/dotnet/docfx/pull/4702#discussion_r293179060) [](ancestors = 293179060)